File: eMovieFinder/eMovieFinder.Model/Dtos/Requests/Actor/ActorInsertRequest.cs
using System;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;

namespace eMovieFinder.Model.Dtos.Requests.Actor
{
    public class ActorInsertRequest
    {
        [Required(ErrorMessage = "FirstName is a Required field")]
        public string FirstName { get; set; }
        [Required(ErrorMessage = "LastName is a Required field")]
        public string LastName { get; set; }
        [Required(ErrorMessage = "ImagePlainText is a Required field")]
        public string? ImagePlainText { get; set; }
        [DefaultValue(null)]
        public byte[]? Image { get; set; }
        [Required(ErrorMessage = "BirthDate is a Required field")]
        public DateTime BirthDate { get; set; }
        [Required(ErrorMessage = "CountryId is a Required field")]
        public int CountryId { get; set; }
        [Required(ErrorMessage = "IMDbLink is a Required field")]
        public string IMDbLink { get; set; }
        [Required(ErrorMessage = "Biography is a Required field")]
        public string Biography { get; set; }
    }
}

File: eMovieFinder-UI/emovie_finder_desktop/lib/src/models/dtos/requests/actor/actor_insert_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'actor_insert_request.g.dart';

@JsonSerializable()
class ActorInsertRequest {
  final String firstName;
  final String lastName;
  final String imagePlainText;
  final DateTime? birthDate;
  final int countryId;
  final String imDbLink;
  final String biography;

  ActorInsertRequest({
    required this.firstName,
    required this.lastName,
    required this.imagePlainText,
    required this.birthDate,
    required this.countryId,
    required this.imDbLink,
    required this.biography
  });

  factory ActorInsertRequest.fromJson(Map<String, dynamic> json) => _$ActorInsertRequestFromJson(json);
  Map<String, dynamic> toJson() => _$ActorInsertRequestToJson(this);
}